client/log_client.go

```diff
@@ -357,6 +357,24 @@ func (c *LogClient) AddSequencedLeaf(ctx context.Context, data []byte, index int
 	return err
 }
 
+// AddSequencedLeaves adds any number of pre-sequenced leaves to the log.
+func (c *LogClient) AddSequencedLeaves(ctx context.Context, dataByIndex map[int64][]byte) error {
+	leaves := make([]*trillian.LogLeaf, 0, len(dataByIndex))
+	for index, data := range dataByIndex {
+		leaf, err := c.BuildLeaf(data)
+		if err != nil {
+			return err
```

Comment:
nit: it might be more helpful to the user if this error message included the leaf index that had a problem